File: 260-SingleNumberIii/260-SingleNumberIii.js
// Last updated: 3/6/2026, 2:09:23 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    let obj = {};
    let res = [];
    for(let char of nums){
        if(obj[char]===undefined){
            obj[char]=1;
        }
        else{
            obj[char]++;
        }
    }
    for(let value in obj){
        if(obj[value]==1){
          res.push(Number(value));
        }
    }
    return res;
};

File: 260-SingleNumberIii/260-SingleNumberIii.cpp
// Last updated: 7/17/2026, 4:27:35 PM
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        unordered_map<int, int> obj;
        vector<int> res;

        for (int charNum : nums) {
            if (obj[charNum] == 0) {
                obj[charNum] = 1;
            } else {
                obj[charNum]++;
            }
        }

        for (auto it : obj) {
            if (it.second == 1) {
                res.push_back(it.first);
            }
        }

        return res;
    }
};